### DISCUSSION TITLE
Breaking change in minor version increase

### DISCUSSION THREAD

**Discussion by wizzymore** [spatie/laravel-health#218] [General]
The pull request with the breaking change: https://github.com/spatie/laravel-health/pull/216

Currently the people that extended the base health model to change the connection are having a breaking change because of this commit.
Before this option was available i was extending the model and setting a custom `$connection` property on the model, as it is normal in Laravel.
Now with this change setting the `$connection` property on the model is completely disregarded and broke my health check on update.

**Comment by freekmurze** [comment]
@eusonlito could you have a look at this?

**Reply by eusonlito** [reply]
@freekmurze, yes, I did not take into account that users would use their own model.

@wizzymore if you set up the connection using the `connection` key in config file does not solve the problem?

**Reply by wizzymore** [reply]
Yes it does. 

Currently this is what i did in my project, but we still need to fix this as I don't think I was the only one using multiple connections and this was the only way to do it before your change. So currently for everyone that did it now their health checks fail because it tries to insert rows in a database that doesn't have the migrations, and if they are not tracking exceptions they will never know that health check is not working for them.

**Reply by wizzymore** [reply]
Have a look at how i fixed it in:

https://github.com/spatie/laravel-health/pull/219

It will work in both methods based on what model you use ( as the provided model doesn't have a connection defined )

**Reply by eusonlito** [reply]
@wizzymore I think it looks good.
